Pytest

Chloe-323 opened the pull request:
merging pytest

Chloe-323 committed:
Sample pytest testing file for mealplan.py

Chloe-323 committed:
Added showcase functions for more functionalities of pytest

Chloe-323 committed:
Added data testing file. Can be imported and used in other tests to
ensure good data.

Chloe-323 committed:
Added one more test case

Chloe-323 committed:
Added testing for missing item in recipe data

Chloe-323 committed:
Work done during meeting

yl5741 committed:
added functional tests and reorganized project layout

yl5741 committed:
Update mealplan.py

yl5741 committed:
functional tests and project layout reorganization

yl5741 committed:
fixing import

Chloe-323 committed:
Fixed MY tests and github test.yml file

Chloe-323 committed:
Testing GH actions

Chloe-323 committed:
Testing GH actions

NativeVex commented:
hey you forgot to use the proper gitignore and are accidentally pushing private ssh keys to the repository

NativeVex committed:
doesn't run perfect, but doesn't have any issues with not seeing the local package

deepsource-autofix[bot] committed:
Format code with black, yapf, autopep8 and isort

NativeVex committed:
Let's see if I was right -- pip -> pipenv

Removed erroneously included .vagrant folder, and migrated testing gh action to
use pipenv to remain in sync with developer environments

NativeVex committed:
Make sure pipenv installs

NativeVex committed:
pytest should be in the global namespace...

NativeVex committed:
maybe order matters a lot?

NativeVex committed:
Gonna install act into my vagrant soon, promise.

NativeVex committed:
I was thinking I had used the --system flag

NativeVex committed:
Less fancy

NativeVex committed:
Lol we caught an actual error

NativeVex committed:
intermediate vagrant troubles

NativeVex committed:
vagrant works better on bionic

Chloe-323 committed:
Redoing this because SOMEONE broke my branch: added component testing
and changed the standard to make ALL numerical values in
nutritional_values floats.

deepsource-autofix[bot] committed:
Format code with yapf, black, autopep8 and isort

NativeVex committed:
last minute fixes

NativeVex committed:
tied up any changes made on infra_cleanup

NativeVex committed:
tied up any changes on sequestered_improvements

NativeVex committed:
not sure what I'm committing

NativeVex committed:
Merge branch 'infra_cleanup' into pytest

NativeVex committed:
workflow must've been incorrectly merged

deepsource-autofix[bot] committed:
Format code with black, yapf, autopep8 and isort

NativeVex committed:
post merge

deepsource-autofix[bot] committed:
Format code with black, yapf, autopep8 and isort

Chloe-323 committed:
Changed GH actions test

NativeVex committed:
fixed path for gh actions

NativeVex committed:
Pipfile.lock is needed for correct packages since we're using --deploy

Chloe-323 committed:
Fixed docker container to use pipenv; currently does not exit with ^C
